src/native/fs/fs_napi.cpp

```diff
@@ -219,6 +230,258 @@ struct Stat : public FSWorker
     }
 };
 
+/**
+ * Getxattr is an fs op
+ */
+struct Getxattr : public FSWorker
+{
+    std::string _path;
+    std::vector<KeyValue> _md;
+    Getxattr(const Napi::CallbackInfo& info) 
+        : FSWorker(info)
+        , _md(0)
+    {
+        _path = info[1].As<Napi::String>();
+        Begin(XSTR() << "Getxattr " << DVAL(_path));
+    }
+    virtual void Work()
+    {
+        ssize_t buf_len, key_len, value_len;
+        char *buf, *key, *val;
+
+        #if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
+                buf_len = listxattr(_path.c_str(), NULL, 0, 0);
+        #else
+                buf_len = listxattr(_path.c_str(), NULL, 0);
+        #endif
+
+        if (buf_len <= 0) {
+            if (buf_len == -1) {
```

Comment:
What is the else case? Why is it just ignored? Even if this code is correct, it is not trivial and requires a comment.

Reply:
I think you meant to identify the `buf_len == 0` case which means there are no xattrs at all?
Better be explicit about it.